File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/impl/ValidatingResult.scala
package ru.mipt.acsl.decode.model.domain.impl

import ru.mipt.acsl.decode.model.domain.pure.ValidatingResult
import ru.mipt.acsl.decode.modeling.aliases.ValidatingMessage

/**
  * @author Artem Shein
  */
object ValidatingResult {
  def empty: ValidatingResult = Seq.empty[ValidatingMessage]
}

File: model/src/main/java/ru/mipt/acsl/decode/model/ValidatingResult.java
package ru.mipt.acsl.decode.model;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Artem Shein
 */
public interface ValidatingResult {

    static ValidatingResult newInstance() {
        return new ValidatingResultImpl(new ArrayList<>());
    }

    boolean add(Message message);

    boolean addAll(ValidatingResult r);

    List<Message> messages();
}
